ldaptive/src/main/java/org/ccci/idm/user/ldaptive/dao/LdaptiveUserDao.java

```diff
@@ -496,6 +506,117 @@ public void removeFromGroup(User user, Group group) throws DaoException {
+        } catch (final ExceededMaximumAllowedResultsException e) {
+            // propagate ExceededMaximumAllowedResultsException exceptions
+            throw e;
+        } catch (final DaoException suppressed) {
+            // suppress any other DaoExceptions
+            return Collections.emptyList();
+        }
+
+        return groups;
+    }
+
+    /**
+     * @param groups                    collection to populate with loaded groups
+     * @param filter                    the LDAP search filter to use when searching
+     * @param limit                     the maximum number of results to return, a limit of 0 indicates that all results
+     *                                  should be returned
+     * @param restrictMaxAllowedResults a flag indicating if maxSearchResults should be observed
+     * @return number of users loaded
+     * @throws DaoException
+     */
+    private int enqueueAllByFilter(@Nonnull final Collection<Group> groups, @Nullable BaseFilter filter,
+                                   final int limit, final boolean restrictMaxAllowedResults) throws DaoException {
+
+        filter = filter != null ? filter.and(FILTER_GROUP) : FILTER_GROUP;
+
+        // perform search
+        Connection conn = null;
+        try {
+            conn = this.connectionFactory.getConnection();
+            conn.open();
+            SearchOperation search = new SearchOperation(conn);
+            final SearchRequest request = new SearchRequest(this.groupsBaseSearchDn, filter);
+
+            // calculate the page size based on the provided limit & maxPageSize
+            int pageSize = maxPageSize;
+            if (limit != SEARCH_NO_LIMIT && pageSize > limit) {
+                pageSize = limit;
+            }
+            if (restrictMaxAllowedResults && maxPageSize != SEARCH_NO_LIMIT && pageSize > maxPageSize + 1) {
+                pageSize = maxPageSize + 1;
+            }
+            final PagedResultsControl prc = new PagedResultsControl(pageSize);
+            request.setControls(prc);
+
+            // retrieve results
+            byte[] cookie = null;
+            int processed = 0;
+            do {
+                // execute request
+                prc.setCookie(cookie);
+                cookie = null;
+                final Response<SearchResult> response = search.execute(request);
+
+                // process response
+                final SearchResult result = response.getResult();
+                final Iterator<LdapEntry> entries = result.getEntries().iterator();
+                while ((limit == SEARCH_NO_LIMIT || processed < limit) && (!restrictMaxAllowedResults ||
+                        maxSearchResults == SEARCH_NO_LIMIT || processed < maxSearchResults) && entries.hasNext()) {
+                    LdapEntry ldapEntry = entries.next();
+                    if (groupValueTranscoder != null) {
+                        final Group group = groupValueTranscoder.decodeStringValue(ldapEntry.getDn());
```

Comment:
see: https://github.com/CruGlobal/idm-user-management/blob/master/ldaptive/src/main/java/org/ccci/idm/user/ldaptive/dao/mapper/AbstractUserLdapEntryMapper.java#L366 for how we currently handle it when loading groups within a User object.